### PR TITLE
Supporter-plus nudge

### PR DESCRIPTION
**Pull request by paul-daniel-dempsey**
Nudging component, encouraging single to recurring contributions. 
Untitled 2

Features->
- 2 coloured header lines, 
- close button, 
- body copy containing currencyGlyph and monthly minimum contribution amount
- monthly recurring link, 
- background-image (using GUIM)

3-cohort A/B test for implementation ->
https://support.thegulocal.com/uk/contribute (Nudge NOT displayed)
https://support.thegulocal.com/uk/contribute#ab-singleToRecurring=control (Nudge NOT displayed)
https://support.thegulocal.com/uk/contribute#ab-singleToRecurring=variantA
https://support.thegulocal.com/uk/contribute#ab-singleToRecurring=variantB

unnamed

[**Trello Card**](https://trello.com/c/gnwhW92Q/982-nudging-single-to-recurring-a-b-test-figma-designs-for-implementation)

## Is this an AB test?
- [x] Yes
- [ ] No

### CODE REVIEW

File: support-frontend/assets/components/checkoutNudge/checkoutNudge.tsx
Comment (undergroundquizscene) [inline support-frontend/assets/components/checkoutNudge/checkoutNudge.tsx:141]: What’s the reason this is broken over two lines? I assume the linter has passed it (or even done it), but I’m surprised – I would’ve expected it all on one line.

**Review by undergroundquizscene — APPROVED** [review]
Looks good to me, though I’m not a frontend expert! In particular, I don’t know how to check Optimize, which is mentioned in the description because it’s an AB test – when I follow that link I just end up at the Optimize homepage, I think.

Might be worth waiting for a review from someone with more frontend knowledge, but I’ll leave that up to you.

**Comment by GHaberis** [comment]
>  I don’t know how to check Optimize, which is mentioned in the description because it’s an AB test – when I follow that link I just end up at the Optimize homepage, I think.

Thank you for highlighting this @undergroundquizscene, we're no longer using Optimize to monitor our AB tests, so we should look at removing these instructions from the repo, this can be done in a follow-up PR, but for now these instructions should be ignored.

File: support-frontend/assets/components/checkoutBenefits/checkoutBenefitsList.tsx
Comment (marialani) [inline support-frontend/assets/components/checkoutBenefits/checkoutBenefitsList.tsx:83]: How about something like?

```suggestion
	nudgeTitleCopySection1: string;
	nudgeTitleCopySection2: string;
```
Something that indicates that these are 2 parts of a whole would be great here

File: support-frontend/assets/components/checkoutBenefits/checkoutBenefitsList.tsx
Comment (marialani) [inline support-frontend/assets/components/checkoutBenefits/checkoutBenefitsList.tsx:130]: We could delete this as the supporter plus nudge supersedes this old component

File: support-frontend/assets/components/checkoutBenefits/checkoutBenefitsList.tsx
Comment (marialani) [reply support-frontend/assets/components/checkoutBenefits/checkoutBenefitsList.tsx:100]: Have seen you'd updated this 👍

File: support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListContainer.tsx
Comment (marialani) [inline support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListContainer.tsx:67]: When do we want the thresholdPrice at 0?

File: support-frontend/assets/helpers/abTests/abtestDefinitions.ts
Comment (GHaberis) [inline support-frontend/assets/helpers/abTests/abtestDefinitions.ts:119]: I believe based on the spec for this AB test there should be 3 cohorts: `control`, `variantA` and `variantB`? The copy inside the container will differ `variantA` and `variantB`.

**Comment by paul-daniel-dempsey** [comment]
Wondered that, control is no-nudge displayed at all? Then I need to define a VariantA and VariantB? 33% chance of each?

**Comment by GHaberis** [comment]
> control is no-nudge displayed at all? Then I need to define a VariantA and VariantB? 33% chance of each?

Exactly, `control` is current behaviour, so no-nudge displayed at all.

File: support-frontend/assets/helpers/abTests/abtestDefinitions.ts
Comment (GHaberis) [inline support-frontend/assets/helpers/abTests/abtestDefinitions.ts:135]: I think we should add a `targetPage: pageUrlRegexes.contributions.allLandingPagesAndThankyouPages` to this AB test, this will mean we only assign users on the "contributions" checkout to this AB test (which is what we want)

File: support-frontend/assets/components/checkoutNudge/checkoutNudge.tsx
Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/checkoutNudge/checkoutNudge.tsx:141]: Yes its just too long for the linter am afraid

File: support-frontend/assets/components/checkoutBenefits/checkoutBenefitsList.tsx
Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/checkoutBenefits/checkoutBenefitsList.tsx:83]: Yes better group naming

File: support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListContainer.tsx
Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListContainer.tsx:67]: Its used for the handleButtonClick (event handler line86), it need's a valid threshold price even when the contributionType is ONE_OFF, otherwise I get  'possibly undefined' in handleButtonClick. Can look into a better way however tomorrow.

File: support-frontend/assets/helpers/abTests/abtestDefinitions.ts
Comment (paul-daniel-dempsey) [reply support-frontend/assets/helpers/abTests/abtestDefinitions.ts:119]: Yes, implemented and redescribed correctly on PR

File: support-frontend/assets/helpers/abTests/abtestDefinitions.ts
Comment (paul-daniel-dempsey) [reply support-frontend/assets/helpers/abTests/abtestDefinitions.ts:139]: Ah yes, have done and will use this going forwards to reduce the pages affected by ab-tests.

File: support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListContainer.tsx
Comment (GHaberis) [reply support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListContainer.tsx:71]: Did you get a typescript error prompting you to make the change? This code would have never actually run when the `contributionType` was ONE_OFF because of the return statement `if (isOneOff(contributionType)) { return null }` that's been removed from this module. I'd be very wary of this change. I don't think we should be repurposing the `checkoutBenefitsListContainer.tsx` component to also double up as the nudge component. I'd create a brand new component for the nudge component that only renders when `isOneOff(contributionType)` and the user is in the appropriate AB test variant.

Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListContainer.tsx:71]: Yes, agree this would be much neater and may help or/simplfy the display side effects have seen, will bring up the nudge component out of the benefits list.

File: support-frontend/assets/components/checkoutBenefits/checkoutBenefitsList.tsx
Comment (marialani) [reply support-frontend/assets/components/checkoutBenefits/checkoutBenefitsList.tsx]: Should be good to delete the component as the nudge component replaces the upsell button

File: support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListContainer.tsx
Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListContainer.tsx]: Ok, finished completing this, would you mind having a look it's acceptable? Have asked Maria to look too. Doing this completely got rid of the display side-effects so will remember components go as high as they can in future.

File: support-frontend/assets/components/checkoutNudge/checkoutNudge.tsx
Comment (GHaberis) [inline support-frontend/assets/components/checkoutNudge/checkoutNudge.tsx:128]: Is it common practise to define CSS rules as props within a `styles` object rather than defining them as individual variables? I can see it works, but my concern is ESLint would not warn us if rules within this `styles` object are no longer used, which it would do if they were individual variables? eg. 

```
const styles = { 
  container: css``,
  top: css``
 }
```

VS 

```
const container = css: ``;
const top = css: ``;
```

This isn't a blocker, but it's worth considering/thinking about.

File: support-frontend/assets/components/checkoutNudge/checkoutNudge.tsx
Comment (GHaberis) [inline support-frontend/assets/components/checkoutNudge/checkoutNudge.tsx:129]: Could we not put this conditional on the before the parent `<div css={styles.container}>` and if `displayValid` is `false` we just return `null` rather than an empty div? We'd need to update the return type of `CheckoutNudge` to `JSX.Element | null`. We do this quite often in the code for conditionally rendered components.

File: support-frontend/assets/components/checkoutNudge/checkoutNudgeContainer.tsx
Comment (GHaberis) [inline support-frontend/assets/components/checkoutNudge/checkoutNudgeContainer.tsx:26]: I don't think we should be calling `initAbTests` to get `abParticipations` they should be available in the Redux state in `state.common. abParticipations`.

File: support-frontend/assets/helpers/abTests/abtestDefinitions.ts
Comment (GHaberis) [inline support-frontend/assets/helpers/abTests/abtestDefinitions.ts:136]: We'll need to set `isActive` to `true` to turn it on.

Comment (paul-daniel-dempsey) [reply support-frontend/assets/helpers/abTests/abtestDefinitions.ts]: Ok, turned on now for use on prod

File: support-frontend/assets/components/checkoutNudge/checkoutNudgeContainer.tsx
Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/checkoutNudge/checkoutNudgeContainer.tsx:26]: Ah I see it in Redux now, will do retrieval from Redux now

Comment (GHaberis) [reply support-frontend/assets/components/checkoutNudge/checkoutNudgeContainer.tsx:26]: For reference this is where we assign `abParticipations` it's value when setting up the initial Redux state: https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/helpers/redux/utils/setup.ts#L88.

File: support-frontend/assets/components/checkoutNudge/checkoutNudge.tsx
Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/checkoutNudge/checkoutNudge.tsx]: Yes makes sense, split them out

File: support-frontend/assets/components/checkoutNudge/checkoutNudge.tsx
Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/checkoutNudge/checkoutNudge.tsx]: Completed - Yes returning null much better than an empty div

File: support-frontend/assets/components/checkoutNudge/checkoutNudge.tsx
Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/checkoutNudge/checkoutNudge.tsx]: Completed- yes makes sense for linter to spot, have split them out now

File: support-frontend/assets/helpers/abTests/abtestDefinitions.ts
Comment (paul-daniel-dempsey) [reply support-frontend/assets/helpers/abTests/abtestDefinitions.ts]: Completed - turned on now for use on prod

File: support-frontend/assets/components/checkoutNudge/checkoutNudgeContainer.tsx
Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/checkoutNudge/checkoutNudgeContainer.tsx]: Completed - makes sense to use redux wherever possible, saves repeat getting of the same participations

File: support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListContainer.tsx
Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListContainer.tsx]: Completed

**Review by GHaberis — APPROVED** [review]
Looks good 👍

**Pull request merged**
